packages/form-js-viewer/src/render/components/form-fields/Table.js

```diff
@@ -1,36 +1,405 @@
-const type = 'table';
+import { isDefined, isNumber, isObject, isString } from 'min-dash';
+import { useExpressionEvaluation } from '../../hooks';
+import { useEffect, useState } from 'preact/hooks';
+import Label from '../Label';
+import { formFieldClasses, prefixId } from '../Util';
+import ArrowDownIcon from './icons/ArrowDown.svg';
+import ArrowUpIcon from './icons/ArrowUp.svg';
+import CaretLeftIcon from './icons/CaretLeft.svg';
+import CaretRightIcon from './icons/CaretRight.svg';
+import classNames from 'classnames';
 
+const type = 'table';
 
+/**
+ * @typedef {('asc'|'desc')} Direction
+ *
+ * @typedef Sorting
+ * @property {string} key
+ * @property {Direction} direction
+ *
+ * @typedef Column
+ * @property {string} label
+ * @property {string} key
+ *
+ * @typedef Props
+ * @property {boolean} disabled
+ * @property {Object} field
+ * @property {string} field.id
+ * @property {Array<Column>|void} field.columns
+ * @property {string|void} field.columnsExpression
+ * @property {string|void} field.label
+ * @property {number|void} field.rowCount
+ * @property {string|void} field.dataSource
+ *
+ * @param {Props} props
+ * @returns {import("preact").JSX.Element}
+ */
 export default function Table(props) {
-  return 'Table';
+  const { field, disabled } = props;
+  const {
+    columns = [],
+    columnsExpression,
+    dataSource = '',
+    rowCount,
+    id,
+    label,
+  } = field;
+
+  /** @type {[(null|Sorting), import("preact/hooks").StateUpdater<null|Sorting>]} */
+  const [ sortBy, setSortBy ] = useState(null);
+  const evaluatedColumns = useEvaluatedColumns(
+    columnsExpression || '',
+    columns,
+  );
+  const columnKeys = evaluatedColumns.map(({ key }) => key);
+  const evaluatedDataSource = useExpressionEvaluation(
+    dataSource.startsWith('=') ? dataSource : `=${dataSource}`,
+  );
+  const data = Array.isArray(evaluatedDataSource) ? evaluatedDataSource : [];
+  const sortedData =
+    sortBy === null
+      ? data
+      : sortByColumn(data, sortBy.key, sortBy.direction);
+
+  /** @type {unknown[][]} */
+  const chunkedData = isNumber(rowCount) ? chunk(sortedData, rowCount) : [ sortedData ];
+  const [ currentPage, setCurrentPage ] = useState(0);
+  const currentChunk = chunkedData[currentPage] || [];
+
+
+  useEffect(() => {
+    setCurrentPage(0);
+  }, [ rowCount, sortBy ]);
+
+  if (disabled) {
+    return <DisabledTable { ...props } />;
+  }
+
+  /** @param {string} key */
+  function toggleSortBy(key) {
+    setSortBy((current) => {
+      if (current === null || current.key !== key) {
+        return {
+          key,
+          direction: 'desc',
+        };
+      }
+
+      if (current.direction === 'asc') {
+        return null;
+      }
+
+      return {
+        key,
+        direction: 'asc',
+      };
+    });
+  }
+
+  return (
+    <div class={ formFieldClasses(type) }>
+      {isString(label) && label.length > 0 ? (
```

Comment:
We don't usually have this check here, I think we should not have a unique implementation and either move this to the label renderer or take it as a follow-up.